### PR TITLE
Bug fix/time mismatch

### PR DESCRIPTION
**Pull request by zaclegarssure**
## What?
You cannot create meetups in the past, they have to at least start now.
Also there were basically no limit on the date (start and end) you set for a meetup, this PR solves this.
Now you can only plan a meetUp a week in advance and make it ends 2 days.
Also if you are editing an ongoing meetUp you won't be able to change the start date, same if it is already terminated, you won't be able to change any date (actually you shouldn't even be able to change a meetup once it is done).
Also now when the picker starts, they will display the previous time that was used, and they will only let you input valid dates.

## Tests?
The code is already covered by current tests, but some specific tests could be done.
The picker tests had to be adapted, notably to work with time relative to the start of the test, since now you cannot input arbitrary dates.

## Why 7 and 2 days ?
Those are just arbitrary values for now, we will choose them later.

### CODE REVIEW

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreationViewModel.kt [hunk 1]
```diff
@@ -136,6 +165,12 @@ class MeetUpCreationViewModel @Inject constructor(
         )
         if (uuid == null) {
             // create new meetup
+            // Make sure the meetup start at least now when it is created
+            if (startDate.value!!.isBefore(Calendar.getInstance())) {
```
Comment (Gerussi) [inline app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreationViewModel.kt:169]: Is it sure that startDate.value is not null?

**Review by Gerussi — APPROVED** [review]
Looks good

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreationViewModel.kt [hunk 2]
```diff
@@ -45,6 +48,26 @@ class MeetUpCreationViewModel @Inject constructor(
     val tags: LiveData<Set<Category>> = _tags
     val participantsId: LiveData<List<String>> = _participantsId
 
+    val maxStartDate: Calendar
+        get() {
+            val maxDate = Calendar.getInstance()
+            maxDate.add(Calendar.DAY_OF_MONTH, 7)
```
Comment (TheblueMan003) [inline app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreationViewModel.kt:54]: Hardcored values, should use a constant instead

**Review by TheblueMan003 — APPROVED** [review]
Otherwise Looks good

File: app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreationViewModel.kt (hunk 1 above)
Comment (zaclegarssure) [reply app/src/main/java/com/github/palFinderTeam/palfinder/meetups/activities/MeetUpCreationViewModel.kt:169]: yes, it is set at line 26.

**Pull request merged**